nlhs_tick_data_hungary/network/network_analyzing/node_manipulator.py

```diff
@@ -0,0 +1,205 @@
+        The weights for the new edges are drawn from a uniform sample with 'a' and 'b' parameters. Where 'a' is the
+        minimum and 'b' is the maximum of the weights in the original network.
+
+        :param int new_node: The newly added node ID.
+        """
+        existing_nodes = list(self.simulation_network.nodes - {new_node})
+        neighbors = random.sample(existing_nodes, min(self.config['num_of_connections'], len(existing_nodes)))
+
+        min_weight, max_weight = self.weight_range
+
+        edges = [
+            (new_node, neighbor, {'weight': random.uniform(min_weight, max_weight)})
+            for neighbor in neighbors
+        ]
+
+        self.simulation_network.add_edges_from(edges)
+
+    def _compute_metric(self) -> float:
+        """
+        Computes a given network metric.
+        Currently implemented metrics: 'APL' and 'LCC'.
+
+        :return float: Computed metric value.
+        """
+        analyzer = NetworkAnalyzer(config={}, network=self.simulation_network)
+        metric_methods = {
+            'APL': analyzer.calc_average_path_length,
+            'LCC': analyzer.calc_size_of_largest_connected_component
+        }
+        return metric_methods.get(self.config['metric'], lambda: None)()
+
+    def _compute_lcc(self) -> int:
+        """
+        Computes the size of the largest connected component (LCC) in the network.
+
+        :return int: Size of the largest connected component.
+        """
+        analyzer = NetworkAnalyzer(config={}, network=self.simulation_network)
+        return analyzer.calc_size_of_largest_connected_component()
+
+    def _get_removal_order(self) -> list:
+        """
+        Determines the order of node removal based on centrality measures.
+
+        :return list: Ordered list of nodes to remove.
+        """
+        if self.config['attack_type'] in ['initial_betweenness', 'initial_degree']:
+            analyzer = NetworkAnalyzer(config={}, network=self.simulation_network)
+            centrality_methods = {
+                'initial_betweenness': analyzer.calc_betweenness_centrality,
+                'initial_degree': analyzer.calc_degree_centrality
+            }
+            centrality = centrality_methods.get(self.config['attack_type'], lambda: {})()
+            return sorted(centrality, key=centrality.get, reverse=True)
+
+        return []
+
+    def _select_node_to_remove(self, removal_order: list) -> int | None:
+        """
+        Selects a node for removal based on the attack strategy.
+
+        :param list removal_order: Precomputed list of nodes to remove.
+        :return dict: The node ID to remove.
+        """
+        if self.config['attack_type'] in ['initial_betweenness', 'initial_degree']:
+            while removal_order:
+                node = removal_order.pop(0)
+                if node in self.simulation_network:
+                    return node
+            return None  # If every node has been removed
+
+        attack_methods = {
```

Comment:
Csináljunk egy statikus osztályt ezekkel a függvényekkel, mint metódusok. Ez így nagyon nehezen olvasható, mi is történik.